### PR TITLE
Added SVC classifier

### PR DESCRIPTION
**Pull request by tupini07**
This pull request introduced the SVC classifier and the ability to pull probabilities from it during classification. 

One other notable change included in this pull request is that the `constants.CLASSIFIERS` dictionary no longer maps from `string` to `classifier object` , it now maps from _name options_ of classifiers to the _standardized_ name for said classifier, in this way we don't have to import the actual classifiers into the `constants` file. The correct model is then initialized in `workflow.init_model` based on these `standardized names` 

Implementation of SVC is part of #269

### CODE REVIEW

File: soweego/commons/constants.py [hunk 1]
```diff
@@ -164,15 +162,22 @@
 WIKIDATA_API_SESSION = 'wiki_api_session.pkl'
 SHARED_FOLDER = '/app/shared/'
 
+
 # Supervised classification
+SVC_CLASSIFIER = 'support_vector_machines'
+NAIVE_BAYES_CLASSFIER = 'naive_bayes'
```
Comment (marfox) [inline soweego/commons/constants.py]:
```suggestion
NAIVE_BAYES_CLASSIFIER = 'naive_bayes'
```
Typo detector enabled 😄

File: soweego/commons/constants.py [hunk 2]
```diff
@@ -164,15 +162,22 @@
 WIKIDATA_API_SESSION = 'wiki_api_session.pkl'
 SHARED_FOLDER = '/app/shared/'
 
+
 # Supervised classification
+SVC_CLASSIFIER = 'support_vector_machines'
```
Comment (marfox) [inline soweego/commons/constants.py]:
```suggestion
SVC_CLASSIFIER = 'support_vector_machines'
LINEAR_SVC_CLASSIFIER = 'linear_support_vector_machines'
```
I would recommend not to drop support for `LinearSVC`.

File: soweego/commons/constants.py [hunk 3]
```diff
@@ -164,15 +162,22 @@
...
-    'naive_bayes': NaiveBayesClassifier,
-    'support_vector_machines': SVMClassifier,
-    'nb': NaiveBayesClassifier,  # Shorthand
-    'svm': SVMClassifier,  # Shorthand
-    'single_layer_perceptron': None,
-    'slp': None  # Shorthand
+    'naive_bayes': NAIVE_BAYES_CLASSFIER,
+    'support_vector_machines': SVC_CLASSIFIER,
+    'single_layer_perceptron': PERCEPTRON_CLASSIFIER,
+    'nb': NAIVE_BAYES_CLASSFIER,  # Shorthand
```
Comment (marfox) [inline soweego/commons/constants.py]:
```suggestion
    'nb': NAIVE_BAYES_CLASSIFIER,  # Shorthand
```

File: soweego/commons/constants.py [hunk 4]
```diff
@@ -164,15 +162,22 @@
...
+
 CLASSIFIERS = {
-    'naive_bayes': NaiveBayesClassifier,
-    'support_vector_machines': SVMClassifier,
-    'nb': NaiveBayesClassifier,  # Shorthand
-    'svm': SVMClassifier,  # Shorthand
-    'single_layer_perceptron': None,
-    'slp': None  # Shorthand
+    'naive_bayes': NAIVE_BAYES_CLASSFIER,
+    'support_vector_machines': SVC_CLASSIFIER,
```
Comment (marfox) [inline soweego/commons/constants.py]:
```suggestion
    'support_vector_machines': SVC_CLASSIFIER,
    'linear_support_vector_machines': LINEAR_SVC_CLASSIFIER,
```

File: soweego/commons/constants.py [hunk 5]
```diff
@@ -164,15 +162,22 @@
...
+PERCEPTRON_CLASSIFIER = 'single_layer_perceptron'
+
 CLASSIFIERS = {
-    'naive_bayes': NaiveBayesClassifier,
-    'support_vector_machines': SVMClassifier,
-    'nb': NaiveBayesClassifier,  # Shorthand
-    'svm': SVMClassifier,  # Shorthand
-    'single_layer_perceptron': None,
-    'slp': None  # Shorthand
+    'naive_bayes': NAIVE_BAYES_CLASSFIER,
```
Comment (marfox) [inline soweego/commons/constants.py]:
```suggestion
    'naive_bayes': NAIVE_BAYES_CLASSIFIER,
```

File: soweego/commons/constants.py [hunk 6]
```diff
@@ -164,15 +162,22 @@
...
+    'support_vector_machines': SVC_CLASSIFIER,
+    'single_layer_perceptron': PERCEPTRON_CLASSIFIER,
+    'nb': NAIVE_BAYES_CLASSFIER,  # Shorthand
+    'svm': SVC_CLASSIFIER,  # Shorthand
```
Comment (marfox) [inline soweego/commons/constants.py]:
```suggestion
    'svm': SVC_CLASSIFIER,  # Shorthand
    'lsvm': LINEAR_SVC_CLASSIFIER,  # Shorthand
```

File: soweego/linker/classify.py [hunk 7]
```diff
@@ -94,14 +94,7 @@ def execute(catalog, entity, model, name_rule, threshold, dir_io):
...
 
-        if isinstance(classifier, rl.NaiveBayesClassifier):
-            predictions = classifier.prob(feature_vectors)
-        elif isinstance(classifier, rl.SVMClassifier):
-            predictions = classifier.predict(feature_vectors)
-        else:
-            err_msg = f'Unsupported classifier: {classifier}. It should be one of {set(constants.CLASSIFIERS)}'
-            LOGGER.critical(err_msg)
-            raise ValueError(err_msg)
+        predictions = classifier.prob(feature_vectors)
```
Comment (marfox) [inline soweego/linker/classify.py]:
```suggestion
        predictions = classifier.predict(feature_vectors) if isinstance(classifier, rl.SVMClassifier) else classifier.prob(feature_vectors)
```

`LinearSVC` doesn't emit probability scores: I'm still keeping support for it.

File: soweego/linker/classify.py [hunk 8]
```diff
@@ -130,10 +123,13 @@ def _zero_when_different_names(prediction, wikidata, target):
 
 
 def _add_missing_feature_columns(classifier, feature_vectors):
+
     if isinstance(classifier, rl.NaiveBayesClassifier):
         expected_features = len(classifier.kernel._binarizers)
-    elif isinstance(classifier, rl.SVMClassifier):
+    
+    elif isinstance(classifier, classifiers.SVCClassifier):
```
Comment (marfox) [inline soweego/linker/classify.py]:
```suggestion
    elif isinstance(classifier, (classifiers.SVCClassifier, rl.SVMClassifier)):
```
Keeping support for LinearSVC.

File: soweego/linker/workflow.py [hunk 9]
```diff
@@ -180,17 +180,20 @@ def in_both_datasets(col: str) -> bool:
 
 def init_model(classifier, binarize, number_of_features):
     # TODO expose other useful parameters
-    if classifier is rl.NaiveBayesClassifier:
-        model = classifier(binarize=binarize)
-    elif classifier is rl.SVMClassifier:
-        model = classifier()
-    # FIXME bad hack due to import issues in constants
-    elif classifier is None:
+    if classifier == constants.NAIVE_BAYES_CLASSFIER:
```
Comment (marfox) [inline soweego/linker/workflow.py]:
```suggestion
    if classifier is constants.NAIVE_BAYES_CLASSIFIER:
```
Is there a reason why you compare with `==` and not `is`?

File: soweego/linker/workflow.py [hunk 10]
```diff
@@ -180,17 +180,20 @@ def in_both_datasets(col: str) -> bool:
...
+        model = rl.NaiveBayesClassifier(binarize=binarize)
+
+    elif classifier is constants.SVC_CLASSIFIER:
+        model = classifiers.SVCClassifier()
+
+    elif classifier is constants.PERCEPTRON_CLASSIFIER:
```
Comment (marfox) [inline soweego/linker/workflow.py]:
```suggestion
    elif classifier is constants.LINEAR_SVC_CLASSIFIER:
        model = rl.SVMClassifier()

    elif classifier is constants.PERCEPTRON_CLASSIFIER:
```

**Comment by tupini07** [comment]
Thanks @marfox ! Good idea to keep support for `rl.SVMClassifier`. You suggestions have been added to this PR .

**Pull request merged**